tagent/src/representations.rs

```diff
@@ -38,3 +40,35 @@ pub struct FileUploadRsp {
     pub version: String,
     pub result: String,
 }
+
+// The Error type that can convert to a actix_web::HttpResponse
+#[derive(Debug)]
+pub struct TagentError {
+    message: String,
+    version: String,
+}
+
+impl fmt::Display for TagentError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Error: {}", self.message)
+    }
+}
+
+impl ResponseError for TagentError {
+    fn error_response(&self) -> HttpResponse {
+        let m = &self.message;
+        let v = &self.version;
+        let r = ErrorRsp {
+            status: String::from("error"),
+            message: m.to_string(),
+            version: v.to_string(),
+            result: String::from("none"),
+        };
+        let body = serde_json::to_value(&r).unwrap().to_string();
+        HttpResponse::BadRequest().body(body)
+    }
+}
+
+pub fn make_tagent_error(message: String, version: String) -> TagentError {
```

Comment:
This `make_tagent_error` was my own naming, and I feel it's not up to par. I think the rust way is to make it a `TagentError::new` constructor. Check the branch `dev-wm` and compare it with this one (`git diff dev dev-wm`). Would you be up to change it? If you like it, we can add it to the PR (`git checkout dev; git merge dev-wm`).

Reply:
totally agree!